src/timemachine-heap.c

```diff
@@ -0,0 +1,150 @@
+/* Copyright (C) 2007-2014 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+
+/**
+ * \file
+ *
+ * \author Mat Oldham <mat.oldham@gmail.com>
+ *
+ * Heap related functions for usage within TimeMachine
```

Comment:
The heap looks a lot like the pool/threadpool functions we already use elsewhere, e.g. https://github.com/inliniac/suricata/blob/master/src/util-pool-thread.h and https://github.com/inliniac/suricata/blob/master/src/util-pool.h. Used for TCP ssn pool and segment pools. Did you consider using these?